### ISSUE TITLE
Warning: session_start() Error

### ISSUE THREAD

**Issue by meta4creations** [ThemeFuse/Unyson#1275]
I have read through multiple threads on this, but the solutions are not working. Also, I believe the issue is a browser cache issue as I'll have this error in Chrome, but it won't show up in Firefox or Safari. And then if I clear my browser cache Chrome the errors go away (for awhile).

I do not get the error when I create the session-test.php as mentioned in other threads.

My session.save_path is set to /tmp in my php.ini

This is the error I get on my site in my header:

```Warning: session_start(): open(/tmp/sess_ftfqdt90c95t80r405a8n72ga5, O_RDWR) failed: Permission denied (13) in /home/compatib/public_html/wp-content/plugins/unyson/framework/helpers/class-fw-flash-messages.php on line 241```

```Warning: session_start(): Cannot send session cache limiter - headers already sent (output started at /home/compatib/public_html/wp-content/plugins/unyson/framework/helpers/class-fw-flash-messages.php:241) in /home/compatib/public_html/wp-content/plugins/unyson/framework/helpers/class-fw-flash-messages.php on line 241```

And then in my footer:

```Warning: Unknown: open(/tmp/sess_ftfqdt90c95t80r405a8n72ga5, O_RDWR) failed: Permission denied (13) in Unknown on line 0```

```Warning: Unknown: Failed to write session data (files). Please verify that the current setting of session.save_path is correct (/tmp) in Unknown on line 0```

Are there any suggestions on resolving this? I have had other's comment on this issue on the site, any my solution was to tell them to clear their browser cache, but that is not a long-term solution.

Thanks!

**Comment by meta4creations** [comment]
This is currently a site on a temporary url if that makes any difference. I have not used Unyson long enough to know if it happens on live sites or not.

**Comment by moldcraft** [comment]
Hello,

I still insist this is a server issue not an Unyson problem.

**Comment by meta4creations** [comment]
Ok, it definitely could be. I did talk to my server support last time it happened, but I didn't realize that it was a per browser issue so they weren't seeing it... I think they though I was crazy.

I'll attempt generating a support ticket again to see if they have any advice.

**Comment by meta4creations** [comment]
I contacted my hosting provide, SiteGround, and they were unable to help. There's no way to replicate this issue as it happens randomly. I am currently seeing the errors on the site, but they are not seeing anything. Here is their response:

> I tested to re-produce the same issue on my end but I was not able to re-create it. 
> 
> Although our goals are to provide a high quality hosting environment, our Technical Support Team is specialized in providing assistance in case your script is not accessible due to a problem with your host server or a vital service running on it. Third-party application-related problems such as this are at times beyond the scope of our expertise (as we are not familiar with the functionality, the structure, and specifics of the scripts).
> 
> However, I recommend you to contact the plugin's developer and ask him for further assistance.
> 
> Best Regards,
> 
> Ivan Delev
> Technical Support Team

**Comment by meta4creations** [comment]
I may be getting somewhere slowly. I found this script https://dodona.wordpress.com/2008/08/26/how-can-i-check-if-php-on-my-webserver-supports-sessions/ to test for session capabilities and it's failing.

Could you explain briefly (if you can) why session variables are needed? My site seems to be running fine without being able to use them (supposedly).

Thanks!

**Comment by danyj** [comment]
You need to show the siteground the test result. Sessions are akey for any cms and if they are failing that siteground must fix it   


> PHP 5 Sessions. A session is a way to store information (in variables) to be used across multiple pages. Unlike a cookie, the information is not stored on the users computer.

**Comment by moldcraft** [comment]
It is used to store flash messages (for e.g. contact form errors)

**Comment by meta4creations** [comment]
I have this issue resolved now. Here's some info SiteGround gave me (and the solution for my particular case) in case you need any additional suggestions for other users:

> Your cPanel user is in the chroot and does not have permissions to write to /tmp. I have created a php.ini file in your "public_html" directory and added the following line to it in order to change the path to the tmp folder:
> 
> Code:
> **session.save_path = "/home/pawhuska/public_html/tmp"**
> 
> I then verified that the path is changed via the phpinfo.php file. The following line in the ".htaccess" file will make the file work for subdirectories as well:
> 
> Code:
> **SetEnv PHPRC /home/pawhuska/public_html/php.ini**

Additionally, they created a /tmp directory in my /public_html directory.

**Comment by manuel-84** [comment]
@meta4creations I had the same issue but was google chrome trying to request a no more existing session, so I solved simply by clearing browser cache/cookies
I hope that your public_html/tmp folder is protected from being accessed from the web.

**Comment by meta4creations** [comment]
@manuel-84 Yes, I had realized that clearing the browser cache got rid of the error. But, that's not a great solution when it comes to normal user's of your site. If a basic user sees this error they are not going to realize they need to clear their cache (nor should a website require a user to). They are just going to think that something is wrong with your site.

The solution I posted above worked for me and my hosting setup. I have not seen the error since.
